xblock/core.py

```diff
@@ -1,83 +1,121 @@
 """
-Core classes for the XBlock family.
-
-This code is in the Runtime layer, because it is authored once by edX
-and used by all runtimes.
-
+Base classes for all XBlock-like objects. Used by all XBlock Runtimes.
 """
-from collections import defaultdict
+import copy
+import functools
 import inspect
+import json
+import logging
 import os
 import warnings
+from collections import OrderedDict, defaultdict
 
 import pkg_resources
-
-import xblock.exceptions
-from xblock.exceptions import DisallowedFileError
-from xblock.fields import String, List, Scope
-from xblock.internal import class_lazy
-import xblock.mixins
-from xblock.mixins import (
-    ScopedStorageMixin,
-    HierarchyMixin,
-    RuntimeServicesMixin,
-    HandlersMixin,
-    XmlSerializationMixin,
-    IndexInfoMixin,
-    ViewsMixin,
+from lxml import etree
+from webob import Response
+
+from xblock.exceptions import (
+    DisallowedFileError,
+    FieldDataDeprecationWarning,
+    JsonHandlerError,
+    KeyValueMultiSaveError,
+    XBlockSaveError,
 )
+from xblock.fields import Field, List, Reference, ReferenceList, Scope, String
+from xblock.internal import class_lazy
 from xblock.plugin import Plugin
 from xblock.validation import Validation
 
-# exposing XML_NAMESPACES as a member of core, in order to avoid importing mixins where
-# XML_NAMESPACES are needed (e.g. runtime.py).
-XML_NAMESPACES = xblock.mixins.XML_NAMESPACES
+# OrderedDict is used so that namespace attributes are put in predictable order
+# This allows for simple string equality assertions in tests and have no other effects
+XML_NAMESPACES = OrderedDict([
+    ("option", "http://code.edx.org/xblock/option"),
+    ("block", "http://code.edx.org/xblock/block"),
+])
 
 # __all__ controls what classes end up in the docs.
-__all__ = ['XBlock']
+__all__ = ['XBlock', 'XBlockAside']
+
 UNSET = object()
 
 
-class XBlockMixin(ScopedStorageMixin):
+class _AutoNamedFieldsMetaclass(type):
```

Comment:
xblock.internal.NamedAttributesMetaclass was responsible for sticking `__name__` attributes on any `Nameable` objects defined on its classes. In practice, Field was the only type of Nameable attribute.

_AutoNamedFieldsMetaclass is essentially the same as NamedAttributesMetaclass, but to simplify things, it only just operates on Fields.

_AutoNamedFieldsMetaclass continues to be the metaclass for both XBlock and XBlockAside (although XBlock further specializes it as _HasChildrenMetaclass, defined further down).